### PR TITLE
Added more logging to the shutdown process

### PR DESCRIPTION
**Pull request by nkoenig**
The goal is to make it a little easier to debug the shutdown process.

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: subt_ign/src/GameLogicPlugin.cc [hunk 1]
```diff
@@ -2307,6 +2317,17 @@ void GameLogicPluginPrivate::Finish(const ignition::msgs::Time &_simTime)
...
+        << "- event:\n"
+        << "  id: " << this->eventCounter << "\n"
+        << "  type: recording_complete\n"
+        << "  time_sec: " << _simTime.sec() << "\n"
+        << "  elapsed_real_time: " << realElapsed << "\n"
+        << "  elapsed_sim_time: " << simElapsed << "\n"
+        << "  total_score: " << this->totalScore << std::endl;
+      this->LogEvent(stream.str());
       this->eventCounter++;
     }
```
Comment (AlejoAsd) [inline subt_ign/src/GameLogicPlugin.cc:2332]: Could we add an `else` case to mention that the ROS node is not active? That way we can verify that the block has not been accessed.

Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc:2332]: the `if (this->rosnode)` block ends on line 2310. The new event starting on line 2320 will always output, regardless of ROS usage. That event will be logged in `events.yml`.

Comment (AlejoAsd) [reply subt_ign/src/GameLogicPlugin.cc:2332]: Ah, my bad. My eyes are used to larger indents, or maybe I need some glasses. Thanks!

**Review by AlejoAsd — APPROVED** [review]

File: subt_ign/src/GameLogicPlugin.cc (hunk 1 above)
Comment (AlejoAsd) [reply subt_ign/src/GameLogicPlugin.cc:2332]: Ah, my bad. My eyes are used to larger indents, or maybe I need glasses. Thanks!

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**